### ISSUE TITLE
`kebab` casing rule complains about "single letter" digits

### ISSUE THREAD

**Issue by tsimbalar** [stoplightio/spectral#1795]
**Describe the bug**
Custom rules which enforce `kebab-case` flag string items made of a single character that is a digit. 

**To Reproduce**

1. Given a custom rule like this one in `.spectral.yaml` : 

```yaml
  enum-kebab-case:
    description: 'enum values should be kebab-case'
    message: '{{value}} is an enum member and should be kebab-case. ({{path}})'
    severity: error
    resolved: false
    given: "$.components.schemas..enum[*]"
    then:
      function: casing
      functionOptions:
        type: kebab
```

2. and a `./docs/swagger.yaml` document like this : 

```yaml
components:
    schemas:
        Phq9AnswerValue:
            type: number
            enum:
                - '0'
                - '1'
                - '2'
                - '3'
            description: "The value for a PHQ-9 Questionnaire question.\n- 0 = \"Not at all\"\n- 1 = \"Several days\"\n- 2 = \"More than half the days\"\n- 3 = \"Nearly every day\""
info:
    title: mindset-app-service
    version: 0.1.0
    description: 'API for Mindset App'
    contact: {}
openapi: 3.0.0
paths: {}
servers:
    -
        url: /
```

3. Run this CLI command 

```
npx spectral lint ./docs/swagger.yaml
```

```
OpenAPI 3.x detected

/Users/yc00067/dev/mindset/mindset-app-service/docs/swagger.yaml
  1:1     hint  openapi-tags                   OpenAPI object should have non-empty `tags` array.
 3:25  warning  oas3-unused-components-schema  Potentially unused components schema has been detected.                                      components.schemas.Phq9AnswerValue
 6:19    error  enum-kebab-case                0 is an enum member and should be kebab-case. (#/components/schemas/Phq9AnswerValue/enum/0)  components.schemas.Phq9AnswerValue.enum[0]
 7:19    error  enum-kebab-case                1 is an enum member and should be kebab-case. (#/components/schemas/Phq9AnswerValue/enum/1)  components.schemas.Phq9AnswerValue.enum[1]
 8:19    error  enum-kebab-case                2 is an enum member and should be kebab-case. (#/components/schemas/Phq9AnswerValue/enum/2)  components.schemas.Phq9AnswerValue.enum[2]
 9:19    error  enum-kebab-case                3 is an enum member and should be kebab-case. (#/components/schemas/Phq9AnswerValue/enum/3)  components.schemas.Phq9AnswerValue.enum[3]

✖ 6 problems (4 errors, 1 warning, 0 infos, 1 hint)

```

**Expected behavior**
It should accept `'0'` pr `'1'` as valid `kebab-case` given they are a single character

**Environment (remove any that are not applicable):**
 - Library version: `5.9.2`
 - OS: `macOS 11.3.1`

**Comment by P0lip** [comment]
Hey!
We indeed do not consider such a string a kebab-case compliant, however you can still work around this.
It appears to me that you might use the `pattern` function.

```yaml
rules:
  enum-kebab-case:
    description: 'enum values should be kebab-case'
    message: '{{value}} is an enum member and should be kebab-case. ({{path}})'
    severity: error
    resolved: false
    given: "$.components.schemas..enum[*]"
    then:
      function: pattern
      functionOptions:
        match: "^[a-z0-9]*(?:-[a-z0-9]+)*$"
```

Would that solve your issue?

**Comment by P0lip** [comment]
@tsimbalar have you had a chance to verify the above rule yet?
We're unlikely to address that in the `casing` function itself, as this is not a kebab-case string after all.

**Comment by tsimbalar** [comment]
@P0lip sorry, I forgot to loop back. 

Yes, this seems like a good workaround, so we can probably close this issue, if we consider it is expected. 

I guess my doubt was more of a "why is `'1'` not valid `kebab-case` ?"

**Comment by P0lip** [comment]
> I guess my doubt was more of a "why is '1' not valid kebab-case ?"

It's not considered such because it has no letters.
Most of regular expressions I found in the Internet were similar to the ones we have in Spectral, hence supporting numbers would be rather something non-standard (despite casing not being any standard AFAIK).

In other words, the casing function is expected to be applied to values that do have some letters.
I hope that clarifies.

**Comment by P0lip** [comment]
I'll close the issue for the time being.
I hope the workaround provided is good enough.
Do feel free to reach out here in case it doesn't work or you simply have other concerns / questions.
Happy to help.

**Comment by tsimbalar** [comment]
Thanks !
